### DISCUSSION TITLE
How to prevent setting a random address when scanning for advertisements?

### DISCUSSION THREAD

**Discussion by potto216** [bluez/bluer#39] [General]
Hi, I have  [code](https://github.com/potto216/BlueR-Test/blob/5b6f290070aad67aae10d0674a7e8f4a6a3a726e/src/client.rs#L113) that sets up a Bluetooth LE controller as an advertiser which is the "server" and in the same code sets up another Bluetooth LE controller  that scans for the advertiser which is the "client". I notice that with HCI and over the air captures that BlueR always sets the client which is scanning with a random public address. I cannot figure out in BlueR how to have it use its regular public address instead. How do I do this?  Thanks!

**Comment by surban** [comment]
I haven't seen anything specific in the documentation about that. I would try making the "client" discoverable and see if that helps. If not, you could ask in the BlueZ Slack since this seems to be a bluetoothd issue.

**Reply by potto216** [reply]
The HCI Command LE Set Random Address is still being set with the `set_discoverable(true)` being called. I'll ask in the BlueZ slack and post what I learn.
